### PR TITLE
Polymorphic SenPy

### PR DESCRIPTION
**Pull request by thekeele**
images

* Introduced `SenPy` abstract class
* Models must implement the `load` and `predict` functions
* The model used by `SenPy` is now driven by the clients request
* Models are only loaded on the initial request, additional requests to that model only predict
* This change caused the yaml to be less useful so I removed it
* A makefile was added for installing deps and running the server (will be helpful for future deployment)
* Pinned deps to a version in `requirements.txt` for sanity

### CODE REVIEW

**Comment by rcehemann** [comment]
spaCy scores are not running between -1 and 1:
image

File: app/models/spaCy/spacy.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+  def load(self):
+    nlp = spacy.load("./app/models/spaCy")
+
+    def predict(text):
+      return nlp(text).cats['POSITIVE']
+
+    nlp.predict = predict
+
+    return nlp
```
Comment (rcehemann) [inline app/models/spaCy/spacy.py]: origin of the scoring problem

File: app/models/spaCy/spacy.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+  def predict(self, text):
```
Comment (rcehemann) [inline app/models/spaCy/spacy.py:12]: never gets called!

**Comment by rcehemann** [comment]
after hotfix:
image

**Review by rcehemann — APPROVED** [review]
:moon: mission is a go :rocket:

File: app/models/spaCy/spacy.py (hunk 1 above)
Comment (thekeele) [reply app/models/spaCy/spacy.py]: https://imgflip.com/s/meme/Jackie-Chan-WTF.jpg

File: app/models/spaCy/spacy.py (hunk 2 above)
Comment (thekeele) [reply app/models/spaCy/spacy.py:12]: 😮

**Comment by thekeele** [comment]
Nice work cleaning up my python code 👍 💯 🏅 🍻

**Pull request merged**